src/ucp/wireup/wireup.c

```diff
@@ -1803,5 +1803,44 @@ unsigned ucp_ep_init_flags(const ucp_worker_h worker,
     return flags;
 }
 
+double ucp_wireup_iface_lat_distance_v1(ucp_context_h context,
+                                        const ucs_linear_func_t *latency,
+                                        const ucs_sys_dev_distance_t *distance)
+{
+    return (context->config.ext.proto_enable) ?
+        (latency->c + distance->latency) : latency->c;
+}
+
+double ucp_wireup_iface_lat_distance_v2(ucp_context_h context,
+                                        const ucs_linear_func_t *latency,
+                                        const ucs_sys_dev_distance_t *distance)
+{
+    ucs_linear_func_t lat;
+
+    if (context->config.ext.proto_enable)
+    {
```

Comment:
```c
double tl_latency = ucp_tl_iface_latency(context, latency);

return context->config.ext.proto_enable ? (tl_latency + distance->latency) : tl_latency ;
```

Reply:
Will this have the same result? ucp_tl_iface_latency() applies a linear function on the latency. With proto v1, we add distance latency to latency.c and pass it to that linear function. But the above change will still use the original latency in that function.

Reply:
so maye
```c
ucs_linear_func_t lat = wiface->attr.latency;

if (context->config.ext.proto_enable) {
    lat.c += wiface->distance.latency;
}
return ucp_tl_iface_latency(context, &lat);
```